lib/ansible/modules/system/ufw.py

```diff
@@ -303,13 +319,47 @@ def ufw_version():
         if command == 'state':
             states = {'enabled': 'enable', 'disabled': 'disable',
                       'reloaded': 'reload', 'reset': 'reset'}
-            execute(cmd + [['-f'], [states[value]]])
+            if module.check_mode:
+                if value in ['reloaded', 'reset']:
+                    changed = True
+                else:
+                    ufw_enabled = pre_state.find("active") != -1
+                    if (value == 'disabled' and ufw_enabled) or (value == 'enabled' and not ufw_enabled):
+                        changed = True
+            else:
+                execute(cmd + [['-f'], [states[value]]])
 
         elif command == 'logging':
-            execute(cmd + [[command], [value]])
+            if module.check_mode:
+                extract = re.search(r'Logging: (on|off) \(([a-z]+)\)', pre_state)
+                if extract:
+                    current_level = extract.group(2)
+                    current_on_off_value = extract.group(1)
+                    if value != "off":
+                        if value != "on" and value != current_level:
+                            changed = True
+                    elif current_on_off_value != "off":
+                        changed = True
+                else:
+                    changed = True
```

Comment:
Also, here, `changed` is only set in check-mode, but *not* in regular non-check-mode. Probably the best is to execute the check-mode code everytime, but in case of non-check-mode, also execute the real command.